src/mpi/init/abort.c

```diff
@@ -121,8 +121,8 @@ int MPI_Abort(MPI_Comm comm, int errorcode)
     
     if (!comm_ptr)
     {
-	/* Use comm world if the communicator is not valid */
-	comm_ptr = MPIR_Process.comm_world;
+	/* Use comm self if the communicator is not valid */
+	comm_ptr = MPIR_Process.comm_self;
```

Comment:
I think if you pass in an invalid communicator, the function should return `MPI_ERR_ARG` and not abort anything.

Reply:
Hmm, I'd be reluctant to not abort when the user asks. It just so happens this is not as foolproof as `abort()`, but IMO, if it compiles, it should cause an abort.

Reply:
If they don’t change the error handler, they’re still going to abort. 😀

Reply:
True. I guess what I'm saying is that if a program compiles and reaches `MPI_Abort`, we should at minimum abort that process no matter what. Whatever messages/return codes the user gets may not make sense, but continuing to run seems like a worse option.